src/main/java/de/fraunhofer/aisec/cpg/graph/FieldDeclaration.java

```diff
@@ -88,6 +96,7 @@ public Expression getInitializer() {
 
   public void setInitializer(Expression initializer) {
     if (this.initializer != null) {
+      setIsDefinition(true);
```

Comment:
Ah, I see your point, but in my opinion also variables that are not explicitly initialized are actually "defined" (or to put it the other way around there is no difference between declaring and defining a variable).

If it has no initialiser it might have a default value, such as `0` or `null`.

Reply:
Or even worse, undefined values that can indeed be used later on. This is in contrast to functions that can't be called when there is no definition. So I would prefer regarding fields as always being auto-defined, removing the distinction between declaration and definition.